src/Sdk/WebApi/WebApi/Results/ResultsHttpClient.cs

```diff
@@ -0,0 +1,130 @@
+            var jsonResponse = await ReadJsonContentAsync<CreateStepSummaryMetadataResponse>(response, cancellationToken);
+            if (!jsonResponse.ok)
+            {
+                throw new Exception($"Failed to mark step summary upload as complete, status code: {response.StatusCode}, ok: {jsonResponse.ok}, jobId: {jobId}, planId: {planId}, stepId: {stepId}, size: {size}, timestamp: {timestamp}");
+            }
+        }
+
+        public static async Task<HttpResponseMessage> UploadFileAsync(string url, string blobStorageType, FileStream file, CancellationToken cancellationToken) {
+            // Upload the file to the url
+            var httpClient = new HttpClient();
+            var request = new HttpRequestMessage(HttpMethod.Put, url)
+            {
+                Content = new StreamContent(file)
+            };
+
+            if (blobStorageType == BlobStorageTypes.AzureBlobStorage) {
+                request.Content.Headers.Add("x-ms-blob-type", "BlockBlob");
+            }
+
+            using (var response = await httpClient.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, cancellationToken)) {
+                if (!response.IsSuccessStatusCode) {
+                    throw new Exception($"Failed to upload file, status code: {response.StatusCode}, reason: {response.ReasonPhrase}, blobStorageType: {blobStorageType}");
+                }
+                return response;
+            }
+        }
+
+        // Handle file upload for step summary
+        public async Task UploadStepSummaryAsync(string jobId, string planId, string stepId, string file, CancellationToken cancellationToken)
+        {
+            try {
+                // Get the upload url
+                var uploadUrlResponse = await GetStepSummaryUploadUrlAsync(jobId, planId, stepId, cancellationToken);
+
+                // Do we want to throw an exception here or should we just be uploading/truncating the data
+                var fileSize = new FileInfo(file).Length;
+                if (fileSize > uploadUrlResponse.soft_size_limit)
+                {
+                    throw new Exception($"File size is larger than the upload url allows, file size: {fileSize}, upload url size: {uploadUrlResponse.soft_size_limit}, blobType: {uploadUrlResponse.blob_storage_type}");
+                }
+
+                // Upload the file
+                using (var fileStream = new FileStream(file, FileMode.Open, FileAccess.Read, FileShare.Read, 4096, true))
+                {
+                    var response = await UploadFileAsync(uploadUrlResponse.summary_url, uploadUrlResponse.blob_storage_type, fileStream, cancellationToken);
+                }
+
+                // Send step summary upload complete message
+                await StepSummaryUploadCompleteAsync(jobId, planId, stepId, fileSize, cancellationToken);
+            }
+            catch (Exception ex) {
+                throw new Exception("Failed to upload step summary to results", ex);
```

Comment:
will this print out a nest error message with tons of stack about the inner exception?